packages/packages/v1-adapters/src/hooks/use-route-status.ts

```diff
@@ -0,0 +1,32 @@
+import useIsPreviewMode from './use-is-preview-mode';
+import useIsRouteActive from './use-is-route-active';
+import { RouteEventDescriptor } from '@elementor/v1-adapters';
+
+type Options = {
+	blockOnKitRoutes?: boolean,
+	blockOnPreviewMode?: boolean,
+}
+
+export default function useRouteStatus(
+	route: RouteEventDescriptor['name'],
+	{
+		blockOnKitRoutes = true,
+		blockOnPreviewMode = true,
+	}: Options = {}
```

Comment:
Why not just one type for all?

Reply:
?, there is type for the options and one for the first argument